zerver/lib/test_helpers.py

```diff
@@ -199,12 +200,21 @@ def stdout_suppressed() -> Iterator[IO[str]]:
 
 def reset_emails_in_zulip_realm() -> None:
     realm = get_realm("zulip")
-    do_set_realm_property(
-        realm,
+    realm_user_default = RealmUserDefault.objects.get(realm=realm)
+    do_set_realm_user_default_setting(
+        realm_user_default,
         "email_address_visibility",
-        Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE,
+        RealmUserDefault.EMAIL_ADDRESS_VISIBILITY_EVERYONE,
         acting_user=None,
     )
+    users = UserProfile.objects.filter(realm=realm)
+    for user in users:
+        do_change_user_setting(
+            user,
+            "email_address_visibility",
+            UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE,
+            acting_user=None,
+        )
```

Comment:
This function seems totally crazy -- it looks like it is primarily called as the first step in a test ... which should be a situation where it's not going to do anything, because we're starting fresh after rolling back the transaction for the previous test. So I think with likely minimal adjustment, we can delete it and just remove it from all its callers.

That can be a separate PR or a prep commit in this one depending what feels convenient.

Reply:
By default the `email_address_visibility` for all the users is set to `ADMINS_ONLY` for the development and test database.

Reply:
I see. I think `ADMINS_ONLY` being the default makes sense - we want to test general code paths with the possibility that users don't have access to email addresses, since that's the more complex case.

The solution is probably to just rename this function and/or give it a docstring explaining what we're doing and why.